src/python/pants/option/arg_splitter.py

```diff
@@ -188,73 +114,25 @@ def add_goal(scope: str) -> str:
                     f"The {si.deprecated_scope} goal was renamed to {si.subsystem_cls.options_scope}",
                 )
 
+            if (si.is_builtin or si.is_auxiliary) and (
```

Comment:
The Rust side doesn't (yet) know about builtin or auxiliary goals, so we need this logic here. Note that it's copied directly from the deleted block above.